src/local/h2_adjust/units.py

```diff
@@ -1,15 +1,183 @@
+
+        result = ts * scale
+        result["unit"] = target_unit
+        return result
+
+    return intensities.apply(_convert_intensities)
+
+
+def h2_mass_factor(species: str):
+    """
+    Calculate a scaling factor to convert 1 kg of H into x kg of `species`
+
+    This assumes that the mass of H is preserved during the conversion.
+    """
+    if species == "H2":
+        # H2 -> 2H
+        factor = get_mass_equivalence(
+            molar_mass_a=2,
+            molar_mass_b=1,
+            stoichiometric_coefficient_a=1,
+            stoichiometric_coefficient_b=2,
+        )
+    elif species == "CH4":
+        # CO2 + 4H2 -> CH4 + 2H20 (over catalysts)
+        factor = get_mass_equivalence(
+            molar_mass_a=2,
+            molar_mass_b=12 + 4,
+            stoichiometric_coefficient_a=4,
+            stoichiometric_coefficient_b=1,
+        )
+    elif species == "NH3":
+        # 3H2 + N2 -> 2NH3 (Haber Bosch process)
+        factor = get_mass_equivalence(
+            molar_mass_a=1,
+            molar_mass_b=14 + 3,
+            stoichiometric_coefficient_a=3,
+            stoichiometric_coefficient_b=1,
+        )
+    else:
+        raise ValueError(species)
+
+    return factor
+
+
+def get_mass_equivalence(
+    molar_mass_a: float,
+    molar_mass_b: float,
+    stoichiometric_coefficient_a: int,
+    stoichiometric_coefficient_b: int,
+) -> float:
+    """
+    Get mass equivalence
+
+    This allows you to easily get the mass equivalence between two species
+    based on their molar mass and an assumption about the chemical reaction
+    used to go between them.
+
+    Examples
+    --------
+    >>> # assume that C <-> CO2 via C + O2 <-> CO2
+    >>> get_mass_equivalence(12, 12 + 2 * 16, 1, 1)
+    44 / 12
+
+    >>> # assume that 3H2 + 2N -> 2NH3
+    >>> get_mass_equivalence(2, 14, 3, 2)
+    14 / 2 * 2 / 3 = 14 / 3
+
+    >>> # the inclusion of stoichiometric ratios is why
+    >>> # it doesn't really matter if it's H or H2
+    >>> # assume that 3H + N -> NH3
+    >>> get_mass_equivalence(1, 14, 3, 1)
+    14 / 3
+    """
+    return (
+        molar_mass_b
+        / molar_mass_a
+        * stoichiometric_coefficient_b
+        / stoichiometric_coefficient_a
+    )
+
+
+def sanitize_production_intensity_units(
+    intensities: scmdata.ScmRun, mass_unit: str = "kg"
+) -> scmdata.ScmRun:
+    """
+    Convert production emissions intensities and leakage rates to a common set of units
+
+    Default target unit is "kg Product / kg H2"
+    """
+
+    def _convert_intensities(ts: scmdata.ScmRun):
+        product: str = ts.get_unique_meta("product", True)
+        carrier: str = ts.get_unique_meta("carrier", True)
+        unit: str = ts.get_unique_meta("unit", True)
+
+        target_unit = (
+            f"{mass_unit} {product if product != 'H2' else 'H'} / {mass_unit} H"
+        )
+
+        try:
+            if unit in ["% of H2", "% H2 component of fuel"]:
```

Comment:
```suggestion
            if unit in ["% of H2", "% H2 component of fuel"]:
            # assume the unit basically means % of H2 that leaks
            # per % of H2 used so the scale is simply 1 / 100 i.e.
            # go from % to kg H2 / kg H2 (i.e. kg H2 leaked/lost per 
            # kg H2 used)
```

Something like that might be best we can do to add context here